localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -784,15 +788,53 @@ geometry_msgs::msg::PoseWithCovarianceStamped NDTScanMatcher::align_using_monte_
 
   output_pose_with_cov_to_log(get_logger(), "align_using_monte_carlo_input", initial_pose_with_cov);
 
-  // generateParticle
-  const auto initial_poses =
-    create_random_pose_array(initial_pose_with_cov, initial_estimate_particles_num_);
+  const double CONVERT_COEFF = std::sqrt(2);
+
+  const auto base_rpy = get_rpy(initial_pose_with_cov);
+  const Eigen::Map<const RowMatrixXd> covariance = {
+    initial_pose_with_cov.pose.covariance.data(), 6, 6};
+  const double stddev_x = CONVERT_COEFF * std::sqrt(covariance(0, 0));
+  const double stddev_y = CONVERT_COEFF * std::sqrt(covariance(1, 1));
+  const double stddev_z = CONVERT_COEFF * std::sqrt(covariance(2, 2));
+  const double stddev_roll = CONVERT_COEFF * std::sqrt(covariance(3, 3));
+  const double stddev_pitch = CONVERT_COEFF * std::sqrt(covariance(4, 4));
+
+  auto uniform_to_normal = [](const double uniform) {
+    assert(-1.0 <= uniform && uniform <= 1.0);
+    constexpr double epsilon = 1.0e-6;
+    const double clamped = std::clamp(uniform, -1.0 + epsilon, 1.0 - epsilon);
+    return boost::math::erf_inv(clamped);
+  };
+
+  auto normal_to_uniform = [](const double normal) { return boost::math::erf(normal); };
+
+  // Optimizing (x, y, z, roll, pitch, yaw) 6 dimensions.
+  // The last dimension (yaw) is a loop variable.
+  const std::vector<bool> is_loop_variable = {false, false, false, false, false, true};
```

Comment:
I think both `roll` and `pitch` are loop variables, too. But since their standard deviations are enough small, there is no problem, right? 
 If so, please describe so in comment.

Reply:
Exactly right.
Added comment.
https://github.com/autowarefoundation/autoware.universe/pull/5078/commits/e484b9f2eea26dc55f413e84e3465fe8a7d6e64e